docs/book/src/images/control-plane-machines-resources.plantuml

```diff
@@ -0,0 +1,44 @@
+@startuml control-plane-machines-resources
```

Comment:
Might be good to rename this to make it clear that this is related specifically to KubeadmControlPlane rather than Control Plane providers in general.

Reply:
I changed to use generic naming for boostrap provider CRDs, such as BarBoostrapConfig and BarBootstrapControlPlane at 95d7f93.

Reply:
Unfortunately, the Control Plane relationships are not as simple to make generic. There is no requirement that a ControlPlane provider operate against Machines or use a Bootstrap Provider. For example, one could create a control plane provider that runs pods within the Management Cluster. I think it would be good to keep this diagram specific for KubeadmControlPlane.

If we wanted to show a generic ControlPlane, the relationships would pretty much end between the Cluster resource and the GenericControlPlane resource.

Reply:
I brought `BarBootstrapControlPlane` back to `KubeadmControlPlane` at 0630a8.

I'll rename this UML title to `kubeadm-control-plane-machines-resources`. What do you think about this name?

Reply:
+1 to the proposed naming

Reply:
I renamed to kubeadm-control-plane-machines-resources at 9427682!